lib/phoenix/socket/transport.ex

```diff
@@ -338,7 +338,7 @@ defmodule Phoenix.Socket.Transport do
         conn
 
       origin_allowed?(check_origin, URI.parse(origin), endpoint, conn) ->
-        conn
+        put_private(conn, :phoenix_origin_checked, true)
```

Comment:
Instead of passing this env var, what if we check `cswsh_check` in this function? If there is no origin or no origin check and `cswsh_check` is set to `:origin`, then the first clause must fail too.

Reply:
Yes this is a good option as well.

We would still need some code in the other module that makes the difference between the 2 cases (CSRF check or no check because it's done here).

A third option if we want to avoid putting a value in the `conn` is to simply call again `Phoenix.Socket.Transport.check_origin/3` in the other module.